wrapper/src/main/java/software/amazon/jdbc/RoundRobinHostSelector.java

```diff
@@ -0,0 +1,182 @@
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import software.amazon.jdbc.util.CacheMap;
+import software.amazon.jdbc.util.Messages;
+import software.amazon.jdbc.util.StringUtils;
+
+public class RoundRobinHostSelector implements HostSelector {
+  public static final AwsWrapperProperty ROUND_ROBIN_HOST_WEIGHT_PAIRS = new AwsWrapperProperty(
+      "roundRobinHostWeightPairs", null,
+      "Comma separated list of database host-weight pairs in the format `<host>:<weight>`.");
+  public static final AwsWrapperProperty ROUND_ROBIN_DEFAULT_WEIGHT = new AwsWrapperProperty(
+      "roundRobinDefaultWeight", "1",
+      "The default weight for any hosts that have not been configured with the `roundRobinHostWeightPairs` parameter.");
+  public static final String STRATEGY_ROUND_ROBIN = "roundRobin";
+  private static final long DEFAULT_ROUND_ROBIN_CACHE_EXPIRE_NANO = TimeUnit.MINUTES.toNanos(10);
+  private static final CacheMap<String, RoundRobinClusterInfo> roundRobinCache = new CacheMap<>();
+
+  public RoundRobinClusterInfo getHostCacheEntry(@NonNull HostSpec host) {
+    return roundRobinCache.get(host.getHost());
+  }
+
+  public synchronized void updateCachePropertiesForHost(
+      @NonNull HostSpec host,
+      @NonNull Properties props) throws SQLException {
+    final RoundRobinClusterInfo clusterInfo = roundRobinCache.get(host.getHost());
+
+    final String defaultWeightString = ROUND_ROBIN_DEFAULT_WEIGHT.getString(props);
+    if (!StringUtils.isNullOrEmpty(defaultWeightString)) {
+      final Integer defaultWeight = Integer.getInteger(defaultWeightString);
+      clusterInfo.defaultWeight = (defaultWeight == null || defaultWeight < 1) ? 1 : defaultWeight;
+    }
+
+    final String hostWeights = ROUND_ROBIN_HOST_WEIGHT_PAIRS.getString(props);
+    if (!StringUtils.isNullOrEmpty(hostWeights)) {
+      final String[] hostWeightPairs = hostWeights.split(",");
+      for (String pair : hostWeightPairs) {
+        final String[] hostWeightPair = pair.split(":");
+        if (hostWeightPair.length != 2
+            || hostWeightPair[0].isEmpty()
+            || hostWeightPair[1].isEmpty()) {
+          throw new SQLException(Messages.get("HostSelector.roundRobinWeightsConfigurationError"));
+        }
+        try {
+          final Integer weight = Integer.valueOf(hostWeightPair[1]);
+          clusterInfo.clusterWeightsMap.put(hostWeightPair[0], weight);
+        } catch (NumberFormatException e) {
+          throw new SQLException(Messages.get("HostSelector.roundRobinInvalidWeight"));
+        }
+      }
+    }
+  }
+
+  private synchronized boolean createCacheEntryForHosts(@NonNull List<HostSpec> hosts) {
```

Comment:
What boolean return means?

Reply:
As I can understand this method initialize the cache with default entries. Can we avoid such default entries?